static/js/controllers/configuration-forms.js

```diff
@@ -28,24 +30,102 @@ var moment = require('moment'),
 
       'ngInject';
 
-      $scope.uploading = false;
+      $scope.json = { uploading: false };
+      $scope.xform = { uploading: false };
 
-      var uploadFinished = function(err) {
-        $scope.uploading = false;
-        $scope.error = !!err;
-        $scope.success = !err;
+      var uploadJsonFinished = function(err) {
+        uploadFinished('json', err);
+      };
+
+      var uploadXFormFinished = function(err) {
+        uploadFinished('xform', err);
+      };
+
+      var uploadFinished = function(type, err) {
+        $scope[type] = {
+          uploading: false,
+          error: !!err,
+          success: !err,
+        };
         if (err) {
           $log.error('Upload failed', err);
         }
       };
 
-      var uploadForms = function() {
-        $scope.uploading = true;
-        $scope.error = false;
-        $scope.success = false;
-        var files = $('#forms-upload-form .uploader')[0].files;
+      var uploadXForm = function() {
+        $scope.xform = {
+          uploading: true,
+          error: false,
+          success: false,
+        };
+
+        var formFiles = $('#forms-upload-xform .form.uploader')[0].files;
+        if (!formFiles || formFiles.length === 0) {
+          uploadXFormFinished(new Error('XML file not found'));
```

Comment:
Return here so it doesn't try to access formFiles[0] and throw an exception (line 73).